src/i18n.tsx

```diff
@@ -1,19 +1,30 @@
 import React from 'react'
 import { i18n } from '@lingui/core'
 import { I18nProvider } from '@lingui/react'
-import { detect, fromUrl } from '@lingui/detect-locale'
+import { detect, fromNavigator } from '@lingui/detect-locale'
 import { ReactNode, useEffect } from 'react'
+import { useLocale } from 'state/user/hooks'
 
-export const locales = ['en', 'pseudo-en', 'de', 'es-AR', 'es-US', 'it-IT', 'iw', 'ro', 'ru', 'vi', 'zh-CN', 'zh-TW']
+//www.codetwo.com/admins-blog/list-of-office-365-language-id/
+export enum Locales {
+  'en' = 'English',
+  'de' = 'Deutsch',
+  'es-AR' = 'español (Argentina)',
+  'es-US' = 'español (Estados Unidos)',
+  'it-IT' = 'italiano',
+  'iw' = 'Hebrew',
+  'ro' = 'română',
+  'ru' = 'русский',
+  'vi' = 'Tiếng Việt',
+  'zh-CN' = '中文 ( 中国 )',
+  'zh-TW' = '中文 ( 台灣 )',
+  'pseudo-en' = '',
+}
 export const defaultLocale = 'en'
 
-const getDetectedLocale = () => {
-  const detected =
-    detect(
-      fromUrl('lang'), // helps local development
-      defaultLocale
-    ) ?? defaultLocale
-  return locales.includes(detected) ? detected : defaultLocale
+export const getNavigatorLocale = (): string | undefined => {
+  const detected = detect(fromNavigator())
```

Comment:
i think we might want to parse the locale from navigator manually.. it seems like we need to split it, uppercase the second part, and fallback to the general one if the region specific version isn't available
https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLanguage/language

Reply:
I agree. I'm tempted to leave it as-is for now. I added a story to the github project to track the work.

Reply:
seems like more work than just doing it, it's hardly a story as much as a 5-10 minute coding task

here's the code from the pkg you're using
https://github.com/lingui/js-lingui/blob/main/packages/detect-locale/src/detectors/fromNavigator.ts

Reply:
wasn't really about the length but staying focused for this iteration. You're right that it's quick though! Added